common-test/src/main/java/io/strimzi/test/JUnitGroupRule.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.test;
+
+import org.junit.Assume;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+public class JUnitGroupRule implements MethodRule {
+
+    @Override
+    public Statement apply(Statement base, FrameworkMethod method, Object target) {
+        JUnitGroup testGroup = method.getAnnotation(JUnitGroup.class);
+
+        if (testGroup == null) {
+            return base;
+        }
+
+        Collection<String> enabledGroups = getEnabledGroups(testGroup.key());
+        Collection<String> declaredGroups = getDeclaredGroups(testGroup);
+
+        if (isGroupEnabled(enabledGroups, declaredGroups)) {
+            return base;
+        }
+
+        return new IgnoreStatement(enabledGroups, declaredGroups);
+    }
+
+    static class IgnoreStatement extends Statement {
+
+        Collection<String> enabledGroups;
+        Collection<String> testGroups;
+
+        public IgnoreStatement(Collection<String> enabledGroups, Collection<String> testGroups) {
+            this.enabledGroups = enabledGroups;
+            this.testGroups = testGroups;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+            Assume.assumeTrue("None of the test groups " + this.testGroups + " are enabled. Enabled test groups: " + this.enabledGroups, false);
+        }
+    }
+
+    private static Collection<String> getEnabledGroups(String key) {
+        Collection<String> enabledGroups = splitProperties(System.getProperty(key));
+
+        return enabledGroups;
+    }
+
+    private static Collection<String> getDeclaredGroups(JUnitGroup testGroup) {
+        String[] declaredGroups = testGroup.value();
+        if (declaredGroups.length != 0) {
+            return new HashSet<>(Arrays.asList(declaredGroups));
+        }
+
+        return Collections.emptyList();
```

Comment:
I would do without the `if` and just return `return new HashSet<>(Arrays.asList(declaredGroups));` whether `declaredGroups` was empty or not.